### PR TITLE
Document (very few) typeclasses

### PR DESCRIPTION
**Pull request by NeQuissimus**
Replacing #1607 with in-repo PR rather than fork. This way we can easily have other people contribute.

The goal of this PR is to document all the (currently existing) type classes into the `examples` module.

- [X] What is a type class? / index
- [ ] Applicative
- [ ] Apply
- [ ] Bind
- [ ] Category
- [ ] Choice
- [ ] Cobind
- [ ] Comonad
- [ ] Compose
- [ ] Contravariant
- [ ] Debug
- [ ] Foldable
- [X] Functor
- [ ] InvariantFunctor
- [ ] IsContravariant
- [ ] IsCovariant
- [ ] Monad
- [ ] Monoid
- [ ] Phantom
- [ ] Profunctor
- [ ] Semigroup
- [ ] Strong
- [ ] Traversable

### CODE REVIEW

**Comment by NeQuissimus** [comment]
I am declaring "PR bankruptcy" on this one... I simply don't get around to working on this... We should still do the little bit of work I have done...

**Comment by jdegoes** [comment]
@NeQuissimus Don't give up yet! 😉

**Comment by NeQuissimus** [comment]
@jdegoes I am not giving up, I am **decomposing** into multiple PRs :D

**Comment by alexknvl** [comment]
:+1: Smaller PRs are better for various reasons IMHO.

File: example/src/main/tut/typeclass/index.md [hunk 1]
```diff
@@ -4,3 +4,81 @@ title:  "Type classes"
...
+val bo = BusinessObject(1234L, 1234567L)
+
+businessObjectDebug.debug(bo)
+```
+
+However, this is not too convenient and very specific to our `BusinessObject`. Would it not be nicer to be able to say `bo.debug`?
+
+Luckily, **Scalaz** makes all this very easy.
+
+We need something similar to the following:
```
Comment (alexknvl) [inline example/src/main/tut/typeclass/index.md]: Not entirely sure if this is the right place to explain syntax enrichment :) Have you considered something along the lines of:

> However, this is not too convenient and very specific to our `BusinessObject`. Would it not be nicer to be able to say `bo.debug`? By default, Scalaz enriches all objects `o` that have an instance of typeclass `Debug` with a method `o.debug`:

and then continue with the `BusinessObject(1234L, 1234567L).debug` snippet.

Comment (alexknvl) [reply example/src/main/tut/typeclass/index.md]: AAh, I see. This is not `Debug` specific, my mistake :smile_cat:

File: example/src/main/resources/microsite/data/menu.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ options:
     - title: Apply
       url: typeclass/Apply.html
       menu_section: typeclass
-    - tiel: Bifunctor
+    - title: Bifunctor
```
Comment (hrhino) [inline example/src/main/resources/microsite/data/menu.yml]: *whistle, whistle* nothing to see here, right?

File: example/src/main/tut/typeclass/index.md [hunk 3]
```diff
@@ -4,3 +4,81 @@ title:  "Type classes"
 ---
 
 # Type classes
+
+_In computer science, a type class is a type system construct that supports ad hoc polymorphism.
```
Comment (hrhino) [inline example/src/main/tut/typeclass/index.md]: "In computer science" is probably implied.

File: example/src/main/tut/typeclass/index.md [hunk 4]
```diff
@@ -4,3 +4,81 @@ title:  "Type classes"
...
+
+Luckily, most of this ships in Scalaz by default.
+
+```tut:reset
+import scalaz.Scalaz._
+import scalaz.typeclass.DebugClass
+
+final case class BusinessObject(id: Long, value: Long)
+
+implicit val businessObjectDebug: Debug[BusinessObject] = instanceOf[DebugClass[BusinessObject]](b => s"BO[${b.id} = ${b.value}]")
```
Comment (hrhino) [inline example/src/main/tut/typeclass/index.md]: at some point we should explain `instanceOf` (but not here).

**Pull request merged**